pymeasure/instruments/tektronix/tek371A.py

```diff
@@ -752,6 +752,23 @@ def initialize(self):
         self.discard_and_disable_all_events()
         time.sleep(1)
 
+    # Wrapper functions for the Adapter object
+    def write(self, command, **kwargs):
+        """Overrides Instrument write method for including write_delay time after the parent call.
+        :param command: command string to be sent to the instrument
+        """
+        actual_write_delay = time.time() - self.last_write_timestamp
+        time.sleep(max(0, self.write_delay - actual_write_delay))
+        super().write(command, **kwargs)
+        self.last_write_timestamp = time.time()
+
+    def ask(self, command):
+        """ Overrides Instrument ask method for including query_delay time on parent call.
+        :param command: Command string to be sent to the instrument.
+        :returns: String returned by the device without read_termination.
+        """
+        return super().ask(command, self.query_delay)
```

Comment:
Do you realise that you will wait `self.query_delay`+`self.write_delay`, as `ask` calls write in the background? Do you really need two different wait intervals, wouldn't the one for `write` be enough?
@bmoneke any feedback? Is there a better approach than this commit (84bdfd16d24d79da7583ae5d9f1621845d373a5d)?

Reply:
I don't know what the instrument needs. However, in principle both delays can be necessary.
The write delay ensures a certain time between two consecutive write commands, while the query delay ensures some delay between writing and reading the answer.

From that, I do not see any issue in the logic.

However, I do see an issue in signature: if "instrument.ask" accepts a command and a "query_delay", the inheriting method should accept both parameters as well.

You can do "super().ask(command, query_delay or self.query_delay)" with a "query_delay = None" parameter.